app/views/topics/show.html.erb

```diff
@@ -0,0 +1,8 @@
+<section>
+  <h1>Topic: <%= @topic.id %></h1>
+
+  <h2> <%= @topic.name %></h2>
+
+  <%= button_to 'Delete Topic', {action: "destroy", id:(@topic.id)},method: :delete %>
```

Comment:
Rails UJS (unobtrusive Javascript) gives you some nice functionality for free. For example, you can easily present a confirmation prompt (just the browser's standard confirm window) as follows:

```
<%= button_to "Delete Topic", { action: "destroy", id: @image.id }, method: :delete, data: { confirm: "Are you sure?" } %>
```

Optional. I'll leave it up to you.